sdk/src/client/api/block_builder/input_selection/requirement/mana.rs

```diff
@@ -16,30 +16,41 @@ impl InputSelection {
 
             // TODO we should do as for the amount and have preferences on which inputs to pick.
             while let Some(input) = self.available_inputs.pop() {
-                selected_mana += input.output.mana();
+                selected_mana += self.total_mana(&input)?;
                 inputs.push(input);
 
                 if selected_mana >= required_mana {
                     break;
                 }
             }
+            if selected_mana < required_mana {
+                return Err(Error::InsufficientMana {
+                    found: selected_mana,
+                    required: required_mana,
+                });
+            }
 
             Ok(inputs)
         }
     }
 
     pub(crate) fn mana_sums(&self) -> Result<(u64, u64), Error> {
-        let required_mana = self.outputs.iter().map(|o| o.mana()).sum::<u64>() + self.mana_allotments;
+        let required_mana =
+            self.outputs.iter().map(|o| o.mana()).sum::<u64>() + self.mana_allotments.values().sum::<u64>();
```

Comment:
Just checking, we don't need to care about auto mana allotment yet at this point?

Reply:
this is run once before allotment and then if an additional allotment is needed

Reply:
Ah I didn't get at first that the allotment requirement would change this field, makes sense now